Lib/contextlib.py

```diff
@@ -239,6 +240,7 @@ async def __aexit__(self, typ, value, traceback):
                     isinstance(value, (StopIteration, StopAsyncIteration))
                     and exc.__cause__ is value
                 ):
+                    exc.__traceback__ = traceback
```

Comment:
needs a test for this case

Reply:
@iritkatriel it doesn't look like this line doesn't do anything. If I delete it I get the same behavior, same with https://github.com/python/cpython/blob/adf2bf3cc554b7c841edce47bb023bbc89d50908/Lib/contextlib.py#L166-L177